WhisperAPI/WhisperAPI/Services/SuggestionsService.cs

```diff
@@ -64,12 +64,24 @@ public bool IsQueryRelevant(SearchQuery searchQuery)
             IEnumerable<SuggestedDocument> coveoIndexDocuments,
             IEnumerable<SearchQuery> queriesList)
         {
-            var agentQueries = queriesList
-                .Where(x => x.Type == SearchQuery.MessageType.Agent)
+            IList<SuggestedDocument> filteredOutChosenSuggestions = new List<SuggestedDocument>();
+            var queries = queriesList
                 .Select(x => x.Query)
                 .ToList();
 
-            return coveoIndexDocuments.Where(x => !agentQueries.Contains(x.Uri));
+            foreach (var coveoIndexDocument in coveoIndexDocuments)
+            {
+                foreach (var query in queries)
```

Comment:
This works, but there is a much cleaner way of doing it. You can do this feature whithout adding a single line of code from the original code. I want you to see the power of Linq, see this method: https://msdn.microsoft.com/en-us/library/bb534972(v=vs.110).aspx

Really, the code that was there before was really close to be working.

Reply:
Fixed